GRDB/Core/Database.swift

```diff
@@ -317,11 +269,30 @@ extension Database {
 extension Database {
 
     // MARK: - Database Setup
-    
+    func setupConnection() throws {
```

Comment:
May I ask you to remove this method, put all its content inside setup() below, and call `configuration.prepareDatabase` *as late as possible*? For example, can we setup foreign keys, busy mode, etc, before the preparation block? My goal is that as much setup as possible is performed before users code runs.

Reply:
I mean that *ideally*, the last lines of `setup()` would be:

```swift
    ... (trace, busyMode, etc)
    // Last step before we can start accessing the database.
    // This is the opportunity to run SQLCipher configuration
    // pragmas such as cipher_page_size, for example.
    try configuration.prepareDatabase?(self)
    try validateFormat() // (This is the static validateDatabaseFormat method that has been turned into an instance method)
}
```